backend/src/apiserver/resource/resource_manager.go

```diff
@@ -399,9 +399,14 @@ func (r *ResourceManager) CreateRun(ctx context.Context, run *model.Run) (*model
 	}
 	// If pipeline version does not exist, create it
 	if pipelineVersion == nil {
-		pipelineVersion, wfTemplate, err = r.createPipelineFromSpecIfNoExisting(manifest, run.Namespace, run.DisplayName, run.Description)
-		if err != nil {
-			return nil, util.Wrap(err, "Failed to create a run due to error fetching pipeline version from pipeline spec")
+		if temp, err := template.New([]byte(manifest)); err != nil {
+			return nil, util.Wrap(err, "Failed to create a run due to error reading manifest")
+		} else {
+			wfTemplate = &temp
+		}
+		pipelineVersion = &model.PipelineVersion{
```

Comment:
I think we can, and probably should, avoid creating this `model.PipelineVersion` object even though this is only in-memory represeentation, because all we need is its manifest part--This is a partially initialized object with `pipelineVersion.PipelineId`, `pipelineVersion. UUID`, and `pipelineVersion.Name` all being empty, and the `pipelineVersion.Description` which is initialized here but not consumed afterwards. 

Following [this description of v1 behavior](https://github.com/kubeflow/pipelines/blame/c4a2a35b5c8ceee8188fab715f348c40ee8b3e00/backend/src/apiserver/resource/resource_manager.go#L685-L690), the code would be cleaner and easier to follow if it's structured as follows:


```golang
if pipelineVersion != nil {
   manifest = pipelineVersion.PipelineSpec
   run.PipelineSpec.PipelineId = pipelineVersion.PipelineId
   run.PipelineSpec.PipelineVersionId = pipelineVersion.UUID
   ...
} else {
   // no need to create `model.PipelineVersion`
   manifest = run.PipelineSpec.PipelineSpecManifest || run.PipelineSpec.WorkflowSpecManifest // this is not valid go syntax, but you got the idea.
   ...
}
```

WDYT?

Reply:
Thank you for taking another look at this! I further simplified the logic and placed it in a separate function as it's used in `CreateJob` method.